socorro/cron/jobs/archivescraper.py

```diff
@@ -0,0 +1,351 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+This is a quick-and-dirty scraper for archive.mozilla.org that optimizes on
+code maintenance followed by the amount of work this needs to do every time it
+runs. It's written with the assumption that it will be temporary.
+
+Rough directory structure::
+
+  pub/
+    firefox/         Firefox builds
+      candidates/    beta, rc, release, and esr builds
+      nightly/       nightly builds
+
+    devedition/      DevEdition (aka Firefox aurora)
+      candidates/    beta builds for Firefox b1 and b2
+
+    mobile/          Fennec builds
+      candidates/    beta, rc, and release builds
+      nightly/       nightly builds
+
+
+This traverses the entire tree every time. If there are missing builds, it will
+pick them up the next time it runs. There's no maintained state for this job
+and it always backfills.
+
+This job only looks for build information for the en-US locale for the first
+platform in a build directory that has build information. Once it's found some
+build information, it moves on to the next version.
+
+This captures the information required to convert a release version into a
+version string. Incoming crash reports have a release version like "63.0", but
+it's really something like "63.0b4" and having the actual version is important
+for analysis.
+
+Since we only do this conversion for aurora, beta, and release versions, we
+don't scrape nightly builds.
+
+Data is stored in the crashstats_productversion table which is managed by the
+webapp (Django).
+
+The record includes the full url of the build file archivescraper pulled the
+information from. This will help for diagnosis of issues in the future.
+
+You can run this in a local development environment like this::
+
+    docker-compose run app shell ./socorro-cmd crontabber \
+        --job=archivescraper --crontabber.class-ArchiveScraperCronApp.verbose
+
+
+The "verbose" argument will cause it to log helpful information for diagnosing
+issues.
+
+"""
+
+from __future__ import print_function
+
+import json
+
+from configman import Namespace
+import psycopg2
+from pyquery import PyQuery as pq
+from six.moves.urllib.parse import urljoin
+
+from socorro.cron.base import BaseCronApp
+from socorro.cron.mixins import using_postgres
+from socorro.lib.requestslib import session_with_retries
+
+
+# Substrings that indicate the thing is not a platform we want to traverse
+NON_PLATFORM_SUBSTRINGS = [
+    'beetmover',
+    'contrib',
+    'funnelcake',
+    'jsshell',
+    'logs',
+    'mar-tools',
+    'partner-repacks',
+    'source',
+    'update'
+]
```

Comment:
These just lets archivescraper skip some of the GET requests that it knows will fail. We could not have this and it'd be fine. It's probably not a comprehensive list.